tests/standalone_tests/test_algorithm_executor.py

```diff
@@ -0,0 +1,132 @@
+import pytest
+from typing import List
+from typing import Dict
+from random import randint
+
+from mipengine.controller.algorithm_executor_helpers import _INode
+from mipengine.controller.algorithm_executor_helpers import TableName
+from mipengine.controller.algorithm_executor import check_same_schema_tables
+
+from mipengine.controller.node_tasks_handler_interface import IQueuedUDFAsyncResult
+
+from mipengine.node_tasks_DTOs import TableSchema
+from mipengine.node_tasks_DTOs import ColumnInfo
+from mipengine.node_tasks_DTOs import DType
+from mipengine.node_tasks_DTOs import TableData
+
+
+class NodeMock(_INode):
+    def __init__(self):
+        self.tables: Dict[str, TableSchema] = {}
+
+    def get_tables(self) -> List[TableName]:
+        pass
+
+    def get_table_schema(self, table_name: TableName):
+        return self.tables[table_name]
+
+    def get_table_data(self, table_name: TableName) -> TableData:
+        pass
+
+    def create_table(self, command_id: str, schema: TableSchema) -> TableName:
+        table_name = f"normal_testnode_cntxtid1_cmdid{randint(0,999)}_cmdsubid1"
+        table_name = TableName(table_name)
+        self.tables[table_name] = schema
+        return table_name
+
+    def get_views(self) -> List[TableName]:
+        pass
+
+    def create_pathology_view(
+        self,
+        command_id: str,
+        pathology: str,
+        columns: List[str],
+        filters: List[str],
+    ) -> TableName:
+        pass
+
+    def get_merge_tables(self) -> List[TableName]:
+        pass
+
+    def create_merge_table(self, command_id: str, table_names: List[TableName]):
+        pass
+
+    def get_remote_tables(self) -> List[str]:
+        pass
+
+    def create_remote_table(
+        self, table_name: str, table_schema: TableSchema, native_node: _INode
+    ):
+        pass
+
+    def queue_run_udf(
+        self, command_id: str, func_name: str, positional_args, keyword_args
+    ) -> IQueuedUDFAsyncResult:
+        pass
+
+    def get_queued_udf_result(
+        self, async_result: IQueuedUDFAsyncResult
+    ) -> List[TableName]:
+        pass
+
+    def get_udfs(self, algorithm_name) -> List[str]:
+        pass
+
+
+@pytest.fixture
+def test_table_schema_a():
+    command_id = "cmndid1"
+    schema = TableSchema(
+        columns=[
+            ColumnInfo(name="var1", dtype=DType.INT),
+            ColumnInfo(name="var2", dtype=DType.STR),
+        ]
+    )
+    return {"command_id": command_id, "schema": schema}
+
+
+@pytest.fixture
+def test_table_schema_b():
+    command_id = "cmndid1"
+    schema = TableSchema(
+        columns=[
+            ColumnInfo(name="var1", dtype=DType.FLOAT),
+            ColumnInfo(name="var2", dtype=DType.INT),
+        ]
+    )
+    return {"command_id": command_id, "schema": schema}
+
+
+def test_check_same_schema_tables(test_table_schema_a, test_table_schema_b):
```

Comment:
you re right, but this whole module is actually kind of a placeholder for the algorithm executor tests that I ll write at some point. There aren't any real tests actually in here yet, I just wanted to see how it would work with the Node mock. Since it is also not accessing the node side it runs instantaneously so I d prefer to let it be as is for now.